Modules/_sqlite/connection.c

```diff
@@ -1442,6 +1457,169 @@ pysqlite_connection_iterdump(pysqlite_Connection* self, PyObject* args)
     return retval;
 }
 
+#ifdef HAVE_BACKUP_API
+static PyObject *
+pysqlite_connection_backup(pysqlite_Connection *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *target = NULL;
+    const char *filename;
+    int pages = -1;
+    PyObject *progress = Py_None;
+    char *name = "main";
+    int rc;
+    int cberr = 0;
+    double sleep_secs = 0.250;
+    sqlite3 *bckconn;
+    sqlite3_backup *bckhandle;
+    static char *keywords[] = {"target", "pages", "progress", "name", "sleep", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|$iOsd:backup", keywords,
+                                     &target, &pages, &progress, &name, &sleep_secs)) {
+        return NULL;
+    }
+
+    if (!PyUnicode_Check(target) && !PyObject_TypeCheck(target, &pysqlite_ConnectionType)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "target argument must be either a string or a Connection instance");
+        return NULL;
+    }
+
+    if (progress != Py_None && !PyCallable_Check(progress)) {
+        PyErr_SetString(PyExc_TypeError, "progress argument must be a callable");
+        return NULL;
+    }
+
+    if (pages == 0) {
+        pages = -1;
+    }
+
+    if (PyUnicode_Check(target)) {
```

Comment:
Is it worth to support path-like objects?

Reply:
Why not indeed... but at that point, should I also change the other functions accepting a path name?

Reply:
On other hand, it may be better to remove support of a file name. The code will be simpler and more reliable if accept only a Connection instance.

Reply:
Uhm, yes, _slightly_ simpler indeed, but that would shift the complication to the caller site: in all my use cases I'd use the file name form, so if that gets dropped I'd need to set up another connection instead...
I added the connection form to deal with Cédric's _opposite_ use case, to _restore_ the content of a test database.
I'm -0 on removing such support, but leave the final decision to you.